### PR TITLE
Add route to extract a frame from a preview

### PR DESCRIPTION
**Pull request by Fabian467**
**Problem**

There is no way to get given from a given preview. While it's an interesting feature to build thumbnails.

**Solution**

Add a route with a preview ID and a frame as parameters that returns the frame as a `.png` file.

### CODE REVIEW

File: zou/app/blueprints/previews/resources.py [hunk 1]
```diff
@@ -102,7 +102,6 @@ def send_movie_file(preview_file_id, as_attachment=False, lowdef=False):
         as_attachment=as_attachment,
     )
 
-
```
Comment (EvanBldy) [inline zou/app/blueprints/previews/resources.py:105]: Please use black to remove this :
```
pip install black 
python -m black .
```
(in your repo folder)

File: zou/app/blueprints/previews/__init__.py [hunk 2]
```diff
@@ -99,6 +100,10 @@
         "/actions/preview-files/<preview_file_id>/set-main-preview",
         SetMainPreviewResource,
     ),
+    (
+        "/data/assets/<instance_id>.png/asset-asset-instances",
```
Comment (EvanBldy) [inline zou/app/blueprints/previews/__init__.py:104]: This route name is not relevant at all, we work on preview files not on assets...
It should be something like : "/data/preview-files/<preview_file_id>/extract-frame"

File: zou/app/blueprints/previews/resources.py [hunk 3]
```diff
@@ -721,6 +720,10 @@ class PreviewFileOriginalResource(BasePreviewPictureResource):
     def __init__(self):
         BasePreviewPictureResource.__init__(self, "original")
 
+class UpdatePreviewThumbnailResource(BasePreviewPictureResource):
```
Comment (EvanBldy) [inline zou/app/blueprints/previews/resources.py:723]: Please remove that it's not used

File: zou/app/blueprints/previews/resources.py [hunk 4]
```diff
@@ -979,7 +982,6 @@ def put(self, entity_id, preview_file_id):
             entity_id, preview_file_id
         )
 
-
```
Comment (EvanBldy) [inline zou/app/blueprints/previews/resources.py:982]: Use black to remove that

File: zou/app/blueprints/previews/resources.py [hunk 5]
```diff
@@ -1005,15 +1007,18 @@ def put(self, preview_file_id):
...
         user_service.check_project_access(task["project_id"])
         user_service.check_entity_access(task["entity_id"])
         asset = entities_service.update_entity_preview(
-            task["entity_id"], preview_file_id
+        task["entity_id"], preview_file_id, 
         )
+
         assets_service.clear_asset_cache(asset["id"])
         shots_service.clear_shot_cache(asset["id"])
+
```
Comment (EvanBldy) [inline zou/app/blueprints/previews/resources.py:1021]: Please remove these modifications

File: zou/app/blueprints/previews/resources.py [hunk 6]
```diff
@@ -1073,7 +1078,7 @@ def put(self, preview_file_id):
 
                     * `updates` that list annotations that needs to be modified.
 
-                    * `deletions` to list the IDs of annotations that needs to be removed.
+                    * `deletions` to CreatePreviewFilePictureResource the IDs of annotations that needs to be removed.
```
Comment (EvanBldy) [inline zou/app/blueprints/previews/resources.py:1081]: Please remove these modifications it's useless

File: zou/app/services/entities_service.py [hunk 7]
```diff
@@ -120,6 +122,11 @@ def update_entity_preview(entity_id, preview_file_id):
         {"%s_id" % entity_type_name: str(entity.id)},
         project_id=str(entity.project_id),
     )
+    events.emit(
+        "%s:update" % entity_type_name,
+        {"%s_id" % entity_type_name: str(entity.id)},
+        project_id=str(entity.project_id)
```
Comment (EvanBldy) [inline zou/app/services/entities_service.py:128]: Please remove these modifications it's useless

File: zou/app/blueprints/previews/__init__.py (hunk 2 above)
Comment (frankrousseau) [reply zou/app/blueprints/previews/__init__.py]: We need the frame number in the route: "/data/preview-files/<preview_file_id>/extract-frame/23".

File: zou/app/services/preview_files_service.py [hunk 8]
```diff
@@ -543,3 +544,33 @@ def get_last_preview_file_for_task(task_id):
         return None
     else:
         return preview.serialize()
+
+def extract_movie_frame(preview_file_id, normalized_movie_path):
```
[the remark this answers is not in the document]
Comment (frankrousseau) [reply zou/app/services/preview_files_service.py]: I don't agree. It applies to movies.

File: zou/utils/movie.py [hunk 9]
```diff
@@ -68,6 +68,58 @@ def generate_thumbnail(movie_path):
...
+            quiet=False
+        )  # voir pour supprimer le fichier après ?
+    except ffmpeg._run.Error as e:
+        log_ffmpeg_error(e, "extracting_frame")
+        raise (e)
+
+    return file_target_path
+
+
+def get_all_frames(movie_path):
```
Comment (frankrousseau) [inline zou/utils/movie.py:99]: It's not related to the pull request. Can you remove it?

**Pull request merged**